src/ActiveLogin.Authentication.BankId.Api/BankIdSimulatedApiClient.cs

```diff
@@ -153,7 +164,7 @@ public async Task<CollectResponse> CollectAsync(CollectRequest request)
             return response;
         }
 
-        private CompletionData GetCompletionData(string endUserIp, CollectStatus status, string personalIdentityNumber)
+        private CompletionData? GetCompletionData(string endUserIp, CollectStatus status, string personalIdentityNumber)
         {
             if (status != CollectStatus.Complete)
```

Comment:
We do double check for CollectionStatus.Complete, if we could move call of GetCompletionData within true case of if statement at line 152 we don't need to check for status here and we can return not nullable CompletionData

Reply:
Is this really possible? We need the CollectResponse even in the false case.